### ISSUE TITLE
Return created/updated objects

### ISSUE THREAD

**Issue by MagicLegend** [Shopify/shopify-php-api#175]
## Overview

The API itself will already return the full object if you do a create/update operation. However, the library swallows this response here:
https://github.com/Shopify/shopify-php-api/blob/0556edf274ceb83346533bd22b662d7622e58c66/src/Rest/Base.php#L60-L74

## Solution

Adding `return $response->getDecodedBody();` at the end of the function should suffice. Note that the `saveAndUpdate` function would also need to be updated, with `return $this->save(true);`.

## Type

- [ ] New feature
- [X] Changes to existing features

## Motivation

> What inspired this feature request? What problems were you facing?

I would like to verify the saved object myself, without needing to query the API again for data that was already returned in the first place.

**Comment by shota** [comment]
+1
I don't know why this has NOT been implemented yet.

**Comment by shota** [comment]
I have created PR for this. This is what should be (`save` return response, `saveAndUpdate` update object itself)
https://github.com/Shopify/shopify-php-api/pull/185

**Comment by Zelfapp** [comment]
This is a pretty serious issues that the [Base.php save()](https://github.com/Shopify/shopify-api-php/blob/0556edf274ceb83346533bd22b662d7622e58c66/src/Rest/Base.php#L60-L74) method returns void. E.g. the [rest api order docs](https://shopify.dev/api/admin-rest/2022-07/resources/order#post-orders) incorrectly state there will be a response. Getting the response with the new order's id is mission critical for our use case, but I don't see a way to access the updated object.

![image](https://user-images.githubusercontent.com/11559805/181652748-5d29a1e2-995c-4191-bd7b-69b16b9eed1b.png)

Any contributors able to post a workaround until this is resolved?

**Comment by MagicLegend** [comment]
I respectfully disagree, and we have my fork running in production right now. It works fine, it just has some quirks that Shopify is not allocating resources to to solve. I prefer using a library that's providing typing over doing it manually in Guzzle, but both will of course work fine.

**Comment by MagicLegend** [comment]
> Still waiting on Shopify...

**Comment by brrrm** [comment]
I fully agree on this issue. I would like to see PR #185 to be committed.

**Comment by brrrm** [comment]
On second thought. When you do for instance: 
```

$product = new Product($session);
$product->title = "my awesome product"

echo $product->id // NULL as it's a fresh unsaved product

// now we save it and set the param $updateObject to true
$product->save($updateObject = true);

echo $product->id // 1234567876 !!!!
// our $product object has been replaced with the object from the response.
```

**Comment by MagicLegend** [comment]
> Yeah, and why is it stale bot? Because Shopify doesn't reply ;)

**Comment by MagicLegend** [comment]
> Yeah, and why is it stale bot? Because Shopify doesn't reply ;)

**Comment by MagicLegend** [comment]
> Yeah, and why is it stale bot? Because Shopify doesn't reply ;)

**Comment by MagicLegend** [comment]
> Yeah, and why is it stale bot? Because Shopify doesn't reply ;)

**Comment by MagicLegend** [comment]
> Yeah, and why is it stale bot? Because Shopify doesn't reply ;)

**Comment by paulomarg** [comment]
Hey folks, thanks for your patience here. We've removed the stalebot from this repo, and I'm going to add this issue to our tracking.

Once again, thank you for your patience!
